rocketpy/motors/TankGeometry.py

```diff
@@ -0,0 +1,464 @@
+import numpy as np
+from copy import copy
+from scipy.optimize import fsolve
+
+
+class TankGeometry:
+    """Class to evaluate mathematical and geometrical properties of a tank
+    geometry, such as volume, mass, center of mass, inertia tensor, etc. The
+    specific geometry is defined by its subclasses and may be used as the main
+    body of the tank or its end caps.
+    """
+
+    def __init__(
+        self, radius, height=None, filled_volume=None, fill_direction="upwards"
+    ):
+        """Initializes a tank geometry object.
+
+        Parameters
+        ----------
+        radius : float
+            Inner radius of the geometry, in meters.
+        height : float, optional
+            Height of the geometry, in meters. If not provided, is set as zero
+            for two-dimenstional geometries or equal to the radius for tri-dimensional.
+        filled_volume : float, optional
+            Volume of the geometry that is filled with propellant, in meters cubed.
+            If not provided, is set as zero.
+        fill_direction : str, optional
+            Direction of propellant filling. Relevant for tri-dimensional geometries
+            which are not vertically simetrical. Can be either "upwards" or "downwards",
+            as such:
+                - "upwards": propellant is filled with the convex site facing down, as in
+                a cup of water. Most used for bottom caps.
+                - "downwards": propellant is filled with the concave site facing down, as
+                in an inverted cup. Most used for upper caps.
+            The default is "upwards", and will be used if the user commits a typing error.
```

Comment:
Seems dangerous to do this. If the user writes "down", it is use "upwards". I believe implementing error handling would be more appropriate here.